wurst/lib/Lodash.wurst

```diff
@@ -0,0 +1,250 @@
+    function call(S s, T t) returns R
+
+public interface TriFunction<S, T, U, R>
+    function call(S s, T t, U u) returns R
+
+public function LinkedList<T>.takeWhile<T>(TriFunction<T, int, LinkedList<T>, bool> predicate) returns LinkedList<T>
+    let list = new LinkedList<T>()
+    let iter = this.iterator()
+    for elem from iter
+        if not predicate.call(elem, list.size(), this)
+            break
+        list.add(elem)
+    iter.close()
+    return list
+
+public function LinkedList<T>.takeWhile<T>(BiFunction<T, int, bool> predicate) returns LinkedList<T>
+    return this.takeWhile((x, y, _z) -> predicate.call(x, y))
+
+public function LinkedList<T>.takeWhile<T>(Predicate<T> predicate) returns LinkedList<T>
+    return this.takeWhile((x, _y) -> predicate.call(x))
+
+public function take<T>(int numElems, LinkedList<T> list) returns LinkedList<T>
+    let result = new LinkedList<T>()
+    let take = min(numElems, list.size())
+    let iter = list.iterator()
+    for elem from iter
+        if result.size() == take
+            break
+        result.add(elem)
+    iter.close()
+    return result
+
+public function take<T>(int numElems) returns Function<LinkedList<T>, LinkedList<T>>
+    return list -> take(numElems, list)
+
+public function LinkedList<T>.take<T>(int numElems) returns LinkedList<T>
+    return take(numElems, this)
+
+public function foldl<Q, T>(Q startValue, BiFunction<T, Q, Q> transform, LinkedList<T> list) returns Q
+    var result = startValue
+    let iter = list.iterator()
+    for elem from iter
+        result = transform.call(elem, result)
+    iter.close()
+    return result
+
+public function foldl<Q, T>(Q startValue, BiFunction<T, Q, Q> transform) returns Function<LinkedList<T>, Q>
+    return list -> foldl(startValue, transform, list)
+
+public function LinkedList<T>.foldl<Q, T>(Q startValue, BiFunction<T, Q, Q> transform) returns Q
+    return foldl(startValue, transform, this)
+
+public function every<T>(Predicate<T> predicate, LinkedList<T> list) returns bool
+    let iter = list.iterator()
+    for elem from iter
+        if not predicate.call(elem)
+            return false
+    iter.close()
+    return true
+
+public function every<T>(Predicate<T> predicate) returns Predicate<LinkedList<T>>
+    return list -> every(predicate, list)
+
+public function LinkedList<T>.every<T>(Predicate<T> predicate) returns bool
+    return every(predicate, this)
+
+public function any<T>(Predicate<T> predicate, LinkedList<T> list) returns bool
+    let iter = list.iterator()
+    for elem from iter
+        if predicate.call(elem)
+            return true
+    iter.close()
+    return false
+
+public function any<T>(Predicate<T> predicate) returns Predicate<LinkedList<T>>
+    return list -> any(predicate, list)
+
+public function LinkedList<T>.any<T>(Predicate<T> predicate) returns bool
+    return any(predicate, this)
+
+public function keys<T, Q>(IterableMap<T, Q> map) returns LinkedList<T>
```

Comment:
Can you iterate over key, value pairs of a hash-map?

Is there any value having an iterator for these?

Reply:
@jonathanmclaus the idea is to build an extensible library, I personally find `key` and `value` functions useful. You can iterate over the keys of an `IterableMap`.